pandas/core/dtypes/cast.py

```diff
@@ -646,6 +648,12 @@ def _maybe_promote(dtype: np.dtype, fill_value=np.nan):
 
         return np.dtype("object"), fill_value
 
+    elif isinstance(dtype, CategoricalDtype):
```

Comment:
Could you move this before the isna check and add the Categorical handling of nan values into this block?

Keeps the Categorical specific code together

Ohterwise this looks good